core/src/main/java/com/linecorp/armeria/common/util/BlockingTaskExecutorBuilder.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import com.linecorp.armeria.common.CommonPools;
+import com.linecorp.armeria.common.Flags;
+
+/**
+ * Provides a builder that is useful for creating an BlockingTaskExecutor.
+ *
+ * @see CommonPools#blockingTaskExecutor()
+ */
+public final class BlockingTaskExecutorBuilder {
+
+    private String threadNamePrefix = "armeria-common-blocking-tasks";
```

Comment:
- Could we also add the following properties:

  ```java
      private boolean daemon;
      private int priority = Thread.NORM_PRIORITY;
      private Function<? super Runnable, ? extends Runnable> taskFunction = Function.identity();
  ```

  .. and their setters? Please see `ThreadFactoryBuilder` to check what these do.
- This is not a common thread pool anymore, so the default value could be `armeria-blocking-tasks`.